### PR TITLE
E2E tester: Allow to optionally create an OIDC token and project

### PR DESCRIPTION
**Pull request by alvaroaleman**
**What this PR does / why we need it**:

Allows to optionally make the e2e tester use the api test code to get a token for the kind setup and makes it create a project then instead of expecting a pre-existing one.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
none
```

### CODE REVIEW

File: api/cmd/conformance-tests/main.go [hunk 1]
```diff
@@ -491,3 +513,40 @@ func shuffle(vals []testScenario) []testScenario {
...
+	if err := wait.PollImmediate(10*time.Second, time.Minute, func() (bool, error) {
+		result, err := client.Project.CreateProject(params, bearerToken)
+		if err != nil {
+			log.Errorw("Failed to create project", "error", fmtSwaggerError(err))
+			return false, nil
+		}
+		projectID = result.Payload.ID
+		return true, nil
+	}); err != nil {
+		return "", fmt.Errorf("failed waiting to project to get successfully created: %v", err)
```
Comment (xrstf) [inline api/cmd/conformance-tests/main.go]: failed waiting *for* project

**Comment by alvaroaleman** [comment]
@xrstf updated the pull, PTAL

**Pull request merged**